src/api/app/models/group.rb

```diff
@@ -34,7 +32,9 @@ class Group < ApplicationRecord
   alias_attribute :name, :title
 
   def self.find_by_title!(title)
-    find_by_title(title) || (raise NotFoundError, "Couldn't find Group '#{title}'")
+    find_by!(title: title)
+  rescue ActiveRecord::RecordNotFound => e
+    raise e, "Couldn't find Group '#{title}'"
```

Comment:
To follow up on my previous comment,  we should also include the backtrace like this:

```suggestion
    raise e, "Couldn't find Group '#{title}'", e.backtrace
```